packages/pwa/app/utils/utils.js

```diff
@@ -158,7 +160,112 @@ export const capitalize = (text) => {
 }
 
 /**
- * A util to return current url configuration
- * @returns {object} - url object from the application configuration file
+ * This function return the identifiers (site and locale) from the given url
+ * The site will always go before locale if both of them are presented in the pathname
+ * @param path {string}
+ * @returns {{siteRef: string, localeRef: string}} - site and locale reference (it could either be id or alias)
  */
-export const getUrlConfig = () => getConfig().app.url
+export const getParamsFromPath = (path) => {
+    const {pathname, search} = new URL(absoluteUrl(path))
+
+    const config = getConfig()
+    const {pathMatcher, searchMatcherForSite, searchMatcherForLocale} = getConfigMatcher(config)
+    const pathMatch = pathname.match(pathMatcher)
+    const searchMatchForSite = search.match(searchMatcherForSite)
+    const searchMatchForLocale = search.match(searchMatcherForLocale)
+
+    // the value can only either in the path or search query param, there will be no overridden
+    const siteRef = pathMatch?.groups.site || searchMatchForSite?.groups.site
+
+    const localeRef = pathMatch?.groups.locale || searchMatchForLocale?.groups.locale
+    return {siteRef, localeRef}
+}
+
+/**
+ * This function returns the url config from the current configuration
+ * @return {object} - url config
+ */
+export const getUrlConfig = () => {
+    const {app} = getConfig()
+    if (!app.url) {
+        throw new Error("Can't find any valid url config. Please check your configuration file.")
```

Comment:
Let's fix up the language in this error a little.

```suggestion
        throw new Error("Cannot find `url` key. Please check your configuration file.")
```

Reply:
I've made the change as suggested.